grails 4 - amazonSNSService - us-east-1 (default) region fails - no signature or method appliable for argument types

grails 4 --  

leaving region as default, or sending in us-east-1 for me results in the error below.  As i read it its saying it wants the argument to be of type "Regions" rather than "Region"


Caused by: groovy.lang.MissingMethodException: No signature of method: com.amazonaws.services.sns.AmazonSNSClientBuilder.withRegion() is applicable for argument types: (com.amazonaws.regions.Region) values: [us-east-1]
Possible solutions: withRegion(com.amazonaws.regions.Regions), withRegion(java.lang.String), getRegion(), setRegion(java.lang.String)
	at grails.plugin.awssdk.sns.AmazonSNSService.afterPropertiesSet(AmazonSNSService.groovy:34)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1837)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1774)

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-version=2.4.2
+version=2.4.3
 awsJavaSdkVersion=1.11.592
 awsKinesisClientVersion=1.10.0
 gradleWrapperVersion=5.5.1

--- a/grails-aws-sdk-cognito/src/main/groovy/grails/plugin/awssdk/cognito/AbstractCognitoAuthService.groovy
+++ b/grails-aws-sdk-cognito/src/main/groovy/grails/plugin/awssdk/cognito/AbstractCognitoAuthService.groovy
@@ -50,7 +50,7 @@ abstract class AbstractCognitoAuthService implements GrailsApplicationAware {
     init() {
         client = AmazonCognitoIdentityClient.builder()
            .withCredentials(AwsClientUtil.buildCredentials(config, cognitoConfig))
-           .withRegion(AwsClientUtil.buildRegion(config, cognitoConfig))
+           .withRegion(AwsClientUtil.buildRegion(config, cognitoConfig).name)
            .build()
     }
 

--- a/grails-aws-sdk-s3/grails-app/services/grails/plugin/awssdk/s3/AmazonS3Service.groovy
+++ b/grails-aws-sdk-s3/grails-app/services/grails/plugin/awssdk/s3/AmazonS3Service.groovy
@@ -11,8 +11,6 @@ import com.amazonaws.services.s3.model.*
 import com.amazonaws.services.s3.transfer.TransferManager
 import com.amazonaws.services.s3.transfer.Upload
 import grails.core.GrailsApplication
-import groovy.transform.stc.ClosureParams
-import groovy.transform.stc.SimpleType
 import groovy.util.logging.Slf4j
 import org.apache.commons.codec.digest.DigestUtils
 import org.springframework.beans.factory.InitializingBean

--- a/grails-aws-sdk-sns/grails-app/services/grails/plugin/awssdk/sns/AmazonSNSService.groovy
+++ b/grails-aws-sdk-sns/grails-app/services/grails/plugin/awssdk/sns/AmazonSNSService.groovy
@@ -32,7 +32,7 @@ class AmazonSNSService implements InitializingBean  {
 
         // Create client
         client = AmazonSNSClientBuilder.standard()
-                .withRegion(region)
+                .withRegion(region.name)
                 .withCredentials(AwsClientUtil.buildCredentials(config, serviceConfig))
                 .withClientConfiguration(AwsClientUtil.buildClientConfiguration(config, serviceConfig))
                 .build()

--- a/groovy-aws-sdk-s3/src/test/groovy/agorapulse/libs/awssdk/s3/S3TransmitSpec.groovy
+++ b/groovy-aws-sdk-s3/src/test/groovy/agorapulse/libs/awssdk/s3/S3TransmitSpec.groovy
@@ -21,7 +21,7 @@ class S3TransmitSpec extends Specification {
         def credentials = new BasicAWSCredentials(accessKey, secretKey)
         def clientConfiguration = AwsSdkUtils.clientConfigurationWithMap([:])
 
-        s3transmit.client = new AmazonS3Client(credentials, clientConfiguration).withRegion(RegionUtils.getRegion(regionName)) as AmazonS3Client
+        s3transmit.client = new AmazonS3Client(credentials, clientConfiguration).withRegion(regionName) as AmazonS3Client
         def result = s3transmit.listBucketNames()
 
         then:

--- a/groovy-aws-sdk-ses/src/main/groovy/com/agorapulse/awssdk/ses/AwsSesMailer.groovy
+++ b/groovy-aws-sdk-ses/src/main/groovy/com/agorapulse/awssdk/ses/AwsSesMailer.groovy
@@ -3,11 +3,15 @@ package com.agorapulse.awssdk.ses
 import com.agorapulse.awssdk.AwsSdkUtils
 import com.amazonaws.AmazonClientException
 import com.amazonaws.AmazonServiceException
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.auth.AWSCredentials
+import com.amazonaws.auth.AWSCredentialsProvider
 import com.amazonaws.auth.BasicAWSCredentials
 import com.amazonaws.regions.Region
 import com.amazonaws.regions.RegionUtils
 import com.amazonaws.services.simpleemail.AmazonSimpleEmailService
 import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClient
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClientBuilder
 import com.amazonaws.services.simpleemail.model.*
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
@@ -35,9 +39,16 @@ class AwsSesMailer {
             log.error("${AmazonSimpleEmailService.ENDPOINT_PREFIX} is not supported in region $regionName")
             return
         }
-        def credentials = new BasicAWSCredentials(accessKey, secretKey)
-        def clientConfiguration = AwsSdkUtils.clientConfigurationWithMap([:])
-        client = new AmazonSimpleEmailServiceClient(credentials, clientConfiguration).withRegion(region) as AmazonSimpleEmailServiceClient
+
+        ClientConfiguration clientConfiguration = AwsSdkUtils.clientConfigurationWithMap([:])
+        client = AmazonSimpleEmailServiceClientBuilder.standard()
+                .withCredentials(new AWSCredentialsProvider() {
+                    @Override AWSCredentials getCredentials() { return new BasicAWSCredentials(accessKey, secretKey) }
+
+                    @Override void refresh() { }
+                })
+                .withClientConfiguration(clientConfiguration)
+                .withRegion(region.name).build()
     }
 
     /**

